lib/BackbeatMetadataProxy.js

```diff
@@ -105,6 +105,58 @@ class BackbeatMetadataProxy extends BackbeatTask {
         });
     }
 
+    headLocation(params, log, cb) {
+        this.retry({
+            actionDesc: 'head object location from source',
+            logFields: {
+                bucket: params.bucket,
+                objectKey: params.objectKey
+            },
+            actionFunc: done => this._headLocationOnce(params, log, done),
+            shouldRetryFunc: err => err.retryable,
+            log,
+        }, cb);
+    }
+
+    _headLocationOnce(params, log, cb) {
+        const { bucket, objectKey, locations } = params;
+        log.debug('heading object location', {
+            where: 'source',
+            bucket, objectKey, locations,
+            method: 'BackbeatMetadataProxy._headLocationOnce',
+        });
+        const cbOnce = jsutil.once(cb);
+        const req = this.backbeatSource.multipleBackendHeadObject({
+            Bucket: bucket,
+            Key: objectKey,
+            Locations: JSON.stringify([{
+                dataStoreName: locations[0].dataStoreName,
+                key: locations[0].key,
+            }]),
+        });
+        attachReqUids(req, log);
+        req.send((err, data) => {
+            if (err) {
+                // eslint-disable-next-line no-param-reassign
+                err.origin = 'source';
+                if (err.ObjNotFound || err.code === 'ObjNotFound') {
+                    return cbOnce(err);
+                }
+                log.error('an error occurred during head object request', {
```

Comment:
could log "head location" instead

Reply:
I kept this as-is because in fact any error would originate from a head object request to the cloud. I think we can find a good medium if I update to say `head object location` as is done above.

Reply:
OK, indeed we could rephrase this way, or "head object on zenko location" or something like this.